scanner/families/malware/clam/clam.go

```diff
@@ -188,7 +195,7 @@ func (s *Scanner) runClamScanWithDaemon(ctx context.Context, fsScanPath string)
 
 func prepareClamDaemon(ctx context.Context, config config.Config) error {
```

Comment:
we should either config as a reference here or switch to a method-based approach (same for `prepareFreshclam`). my suggestion is to switch this to
```suggestion
func (s *Scanner) prepareClamDaemon(ctx context.Context) error {
```

the `New` method can easily be changed to
```go
func New(ctx context.Context, _ string, config types.ScannersConfig) (families.Scanner[*types.ScannerResult], error) {
	scanner := &Scanner{
		config: config.Clam,
	}

	// Prepare freshclam to sync database data
	if err := scanner.prepareFreshclam(ctx); err != nil {
		return nil, fmt.Errorf("failed to run freshclam: %w", err)
	}

	// Prepare clam daemon
	if err := scanner.prepareClamDaemon(ctx); err != nil {
		return nil, fmt.Errorf("failed to run freshclam: %w", err)
	}

	return scanner, nil
}
```